volplugin/volplugin.go

```diff
@@ -32,7 +32,9 @@ type VolumeResponse struct {
 // Daemon starts the volplugin service.
 func Daemon(debug bool, master, host string) error {
 	driverPath := path.Join(basePath, "volplugin.sock")
-	os.Remove(driverPath)
+	if err := os.Remove(driverPath); err != nil && !os.IsPermission(err) {
```

Comment:
We should probably just do a blanket check for the error here and yield if any. If the permissions are invalid, we still won't be able to create the socket file later.